File: robocode2-server/src/main/java/dev/robocode/tankroyale/server/mappers/BulletsToBulletStatesMapper.java
package dev.robocode.tankroyale.server.mappers;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

import dev.robocode.tankroyale.server.model.Bullet;
import net.robocode2.schema.BulletState;

public final class BulletsToBulletStatesMapper {

	private BulletsToBulletStatesMapper() {}

	public static List<BulletState> map(Set<Bullet> bullets) {
		List<BulletState> bulletStates = new ArrayList<>();
		for (Bullet bullet : bullets) {
			bulletStates.add(BulletToBulletStateMapper.map(bullet));
		}
		return bulletStates;
	}
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/mapper/BulletsToBulletStatesMapper.kt
package dev.robocode.tankroyale.server.mapper

import dev.robocode.tankroyale.schema.BulletState
import dev.robocode.tankroyale.server.model.Bullet

object BulletsToBulletStatesMapper {
    fun map(bullets: Set<Bullet>): List<BulletState> {
        val bulletStates = mutableListOf<BulletState>()
        bullets.forEach { bulletStates += BulletToBulletStateMapper.map(it) }
        return bulletStates
    }
}